core/src/main/scala/zio/config/ReadError.scala

```diff
@@ -1,20 +1,11 @@
 package zio.config
 
-sealed trait ReadError[+K, +V] {
-  def key: K
-}
+sealed trait ReadError[+K]
 
 object ReadError {
-  final case class MissingValue[K](key: K)                                extends ReadError[K, Nothing]
-  final case class ParseError[K, V](key: K, provided: V, message: String) extends ReadError[K, V]
-  final case class FatalError[K](key: K, cause: Throwable)                extends ReadError[K, Nothing]
-
-  def missingValue[K, V](key: K): ReadError[K, V] =
-    MissingValue(key)
-
-  def parseError[K, V](key: K, provided: V, message: String): ReadError[K, V] =
-    ParseError(key, provided, message)
-
-  def fatalError[K, V](key: K, cause: Throwable): ReadError[K, V] =
-    FatalError(key, cause)
+  final case class MissingValue[K](key: K, positions: Option[Int] = None)           extends ReadError[K]
+  final case class ParseError[K, V](key: K, message: String)                        extends ReadError[K]
+  final case class Unknown[K](key: K, cause: Throwable)                             extends ReadError[K]
+  final case class OrErrors[K](leftErrors: ReadError[K], rightErrors: ReadError[K]) extends ReadError[K]
+  final case class AndErrors[K](errors: ::[ReadError[K]])                           extends ReadError[K]
```

Comment:
There should be one more called `ListErrors` which is a list of errors dealing with transforming values of a list.
`AndErrors` should then be just `left` and `right` similar to `OrErrors`